### PR TITLE
[SPARK-27090][CORE] Removing old LEGACY_DRIVER_IDENTIFIER ("<driver>")	

### PR DESCRIPTION
**Pull request by shivusondur**
## What changes were proposed in this pull request?
LEGACY_DRIVER_IDENTIFIER and its reference are removed.
corresponding references test are updated.

## How was this patch tested?
tested  UT test cases

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/ui/UIUtilsSuite.scala [hunk 1]
```diff
@@ -122,8 +122,8 @@ class UIUtilsSuite extends SparkFunSuite {
   test("decodeURLParameter (SPARK-12708: Sorting task error in Stages Page when yarn mode.)") {
     val encoded1 = "%252F"
     val decoded1 = "/"
-    val encoded2 = "%253Cdriver%253E"
-    val decoded2 = "<driver>"
+    val encoded2 = "driver"
```
Comment (srowen) [inline core/src/test/scala/org/apache/spark/ui/UIUtilsSuite.scala]: You can delete this check I think; there is nothing to encode

**Review by srowen — COMMENTED** [review]
My only concern here is not being able to read older spark event logs with `<driver>`, I think? but this could be reasonable for Spark 3. I'm OK with this.

File: core/src/test/scala/org/apache/spark/ui/UIUtilsSuite.scala (hunk 1 above)
Comment (shivusondur) [reply core/src/test/scala/org/apache/spark/ui/UIUtilsSuite.scala]: done. 
Removed unnecessary encode.

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
ok to test

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala [hunk 2]
```diff
@@ -197,7 +197,6 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
 
   test("BlockManagerId.isDriver() backwards-compatibility with legacy driver ids (SPARK-6716)") {
```
Comment (attilapiros) [inline core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala]: Please update the test name as from now on it does not test "backwards-compatibility with legacy driver ids".

**Review by attilapiros — COMMENTED** [review]
Minor comment. Otherwise looks good.

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala (hunk 2 above)
Comment (shivusondur) [reply core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala]: @attilapiros 
Changed the testcase name, Thanks for your comment.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**